src/main/java/seedu/financeit/goaltracker/GoalTracker.java

```diff
@@ -30,40 +31,103 @@
+        System.out.println("[1]. expense <amount> for <Month in number form (e.g 01 -> Jan)>");
+        System.out.println("[2]. income <amount> for <Month in number form (e.g 01 -> Jan)>");
+        System.out.println("[3]. To display goals for respective months, enter the following commands"
+                + " display expense/income for <Month in number form (e.g 01 -> Jan)>");
+        System.out.println("[4]. edit expense/income <amount> for <Month in number form (e.g 01 -> Jan)>");
+        System.out.println("[5]. Enter <exit> to leave Goals Tracker");
+
         exitTracker = false;
         input = UiManager.handleInput();
-        cmdPacket = new InputParser().parseInput(input.toLowerCase());
+        cmdPacket = new InputParser().parseGoalCommand(input.toLowerCase());
+        editPacket = new InputParser().parseEditCommand(input.toLowerCase());
         while (!exitTracker) {
-            switch (cmdPacket.getCommandString()) {
+            switch (cmdPacket[0]) {
             case "expense":
                 setExpenseGoal();
                 break;
             case "income":
                 setIncomeGoal();
                 break;
+            case "display":
+                displayGoal();
+                break;
+            case "edit":
+                editGoal();
+                break;
             case "exit":
                 exitTracker = true;
                 break;
             default:
                 System.out.println("Invalid Command");
                 input = UiManager.handleInput();
-                cmdPacket = new InputParser().parseInput(input.toLowerCase());
+                cmdPacket = new InputParser().parseGoalCommand(input.toLowerCase());
                 break;
             }
         }
     }
 
+    public static void editGoal() {
+        try {
+            month = Month.of(Integer.parseInt(editPacket[4]));
+            if (editPacket[1].equals("expense")) {
+                expenseGoal = Integer.parseInt(editPacket[2]);
+                for (int i = 0; i < totalGoalList.getListSize(); i++) {
+                    try {
+                        if (totalGoalList.getGoal().get(i).getExpenseMonth().equals(month)) {
+                            goalSetter = new Goal(expenseGoal, "Expense", month);
+                            totalGoalList.getGoal().set(i, goalSetter);
+                            UiManager.printWithStatusIcon(Constants.PrintType.GOAL_STATUS, "You have successfully"
+                                    + " edited your expense goal for " + month);
+                            main();
+                        } else {
+
+                        }
+                    } catch (NullPointerException e) {
+                        continue;
+                    }
+                }
+
+                UiManager.printWithStatusIcon(Constants.PrintType.ERROR_MESSAGE, "You did not have "
+                        + "any existing goal for " + month);
+                main();
```

Comment:
Avoid calling main() over and over again - this creates a new loop. So if the user does 10 operations, this will cause 10 nested main loops to be created. Should return to the top level main loop directly (i.e. use `return;`) and allow the main loop to continue